pkg/adminapi/client.go

```diff
@@ -100,9 +104,9 @@ func GetKongClientForWorkspace(ctx context.Context, adminURL string, wsName stri
 			return nil, fmt.Errorf("creating workspace: %w", err)
 		}
 	}
-	client, err = kong.NewClient(kong.String(adminURL+"/"+wsName), httpclient)
-	if err != nil {
-		return nil, fmt.Errorf("creating Kong client: %w", err)
-	}
+
+	// ensure that we set the workspace appropriately
+	client.SetWorkspace(wsName)
```

Comment:
```suggestion
	ws, err := client.SetWorkspace(wsName)
	if err != nil {
	}
```

Reply:
Sorry @ccfishk I'm not sure I understand where this suggestion is coming from, as the function has no return values:

https://github.com/Kong/go-kong/blob/26f0e0eec8e8a6e20eb149e051c331e022cc286c/kong/client.go#L171

```
// SetWorkspace sets the Kong Enteprise workspace in the client.
// Calling this function with an empty string resets the workspace to default workspace.
func (c *Client) SetWorkspace(workspace string) {
	c.workspaceLock.Lock()
	defer c.workspaceLock.Unlock()
	c.workspace = workspace
}
```

Reply:
I'm marking this resolved because I don't see that there is any action that can be taken here, but please let me know if there's something that you wouldn't like to see followed up on